trust-quorum/src/validators.rs

```diff
@@ -268,10 +268,17 @@ impl ValidatedReconfigureMsg {
 
         if current_epoch == new_msg.epoch {
             if existing_msg != new_msg {
+                // TODO: This should be an `Alarm`, but I"m not sure how
+                // to handle that. We could include an `Alarm` variant inside
+                // `ReconfigurationError`, but that would require a match/method
+                // to check for an alarm and pull it out. We could also return either
+                // an `Alarm` or a `ReconfigurationError` inside `Result::Err`. This is
+                // probably the best approach, but I'm open to other structures.
```

Comment:
Tricky! Yeah I'd probably return an error enum with `Alarm` and `ReconfigurationError` as variants, and maybe even consider not implementing `std::error::Error` or `fmt::Display` to ensure that people don't accidentally convert it to `anyhow::Error`.